sql/walrus--0.1.sql

```diff
@@ -37,18 +36,22 @@ create type realtime.user_defined_filter as (
 );
 
 
+create function realtime.to_regrole(role_name text)
+    returns regrole
+    immutable
+    language sql
+    -- required to allow use in generated clause
+as $$ select role_name::regrole $$;
+
+
 create table realtime.subscription (
-    -- Tracks which users are subscribed to each table
-    id bigint not null generated always as identity,
-    user_id uuid not null,
-    -- Populated automatically by trigger. Required to enable auth.email()
-    email varchar(255),
+    -- Tracks which subscriptions are active
+    id uuid primary key,
     entity regclass not null,
     filters realtime.user_defined_filter[] not null default '{}',
-    created_at timestamp not null default timezone('utc', now()),
-
-    constraint pk_subscription primary key (id),
-    unique (entity, user_id, filters)
+    claims jsonb not null,
```

Comment:
@olirice what about we default `claims` to `'{}'`?

Reply:
the insert trigger enforces that the claims contents contains a `role` key since its required, so the default would fail to insert

Reply:
@olirice

> anonymous users don't have a jwt

Realtime server receives JWT with the following claims for anonymous user subscribing to a topic:

```
{
  "role": "anon",
  "iat": 1625372973,
  "exp": 1940948973
}
```

So we should be fine adding in a `not null` constraint for `claims_role`.

> what do you think about adding a required parameter like `realtime.apply_rls(..., default_role regrole)`

I like this but the only problem is there's no way right now for devs to pass it in if they're using hosted Supabase and use a different anonymous role (not `anon`). We can of course add an option to the dashboard in the future but for now we can just tell devs to include `role` in JWT.

Reply:
ok awesome, thats a much easier change. The `claims_role` is now not null